static/js/main.js

```diff
@@ -27,6 +38,15 @@ const getToolbox = () => {
 				${extraCategories}
 			</xml>
 		`;
+	} else if(ifCategories) {
```

Comment:
Thank you for commenting this because this layout was one of challenges I spent many hours in order to add conditional blocks because if you look at the tag for the controls_if closely,  there are `<mutation ~~/>` inside the `block`. So I chose your solution having `variable` category to have extraCategories.
Based on your comment, I also tested these categories in different ways this morning. 
If I understand your  explanation correct, I don't think we have many options to place these conditional blocks with other built in blocks under the same category.

Reply:
Thank you for this code.